cli/src/semgrep/metrics.py

```diff
@@ -157,12 +165,43 @@ def configure(
 
         self.metrics_state = metrics_state or MetricsState.AUTO
 
+    # TODO(cooper): It would really be best if EngineType included all of the
+    # information here, but I am a bit concerned about changing it, since it is
+    # currently an enum. Ideally it would be more like osemgrep's Engine_type.t,
+    # but that seems difficult to render here, and would seem to require
+    # threading much more information through that type. Since we only really
+    # care about the additional information being bundeled for metrics, we'll
+    # just take some additional parameters here. Currently this is just for
+    # secrets, but the same would apply for (supply chain)-related information.
     @suppress_errors
-    def add_engine_type(self, engineType: "EngineType") -> None:
+    def add_engine_config(
+        self, engineType: "EngineType", secrets_origins: Optional[str]
```

Comment:
At minimum, needs an explanation of what `secrets_origins` is. Ideally this is structured in a way that makes that easier to discover.

Reply:
Commenting on the other one; can add a docstring here.